### PR TITLE
Pull out separate Jass Parser from Wurst and Jurst grammars

### PR DESCRIPTION
**Pull request by Frotty**
And parse all `.j` files with it.
Fixes https://github.com/wurstscript/WurstScript/issues/702

Maybe in the future we should remove all jass parts from the other grammars?

### CODE REVIEW

**Comment by Cokemonkey11** [comment]
Very cool. I don't plan to review this but let me know if you'd like me to

File: de.peeeq.wurstscript/src/main/antlr/de/peeeq/wurstscript/antlr/Jass.g4 [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    | jassTypeExpr 'array'
+;
+
+exprVarAccess: varname=ID indexes?;
+
+indexes: '[' jassExpr ']';
+
+jassExpr:
+      jassExprPrimary
+    | left=jassExpr op=('*'|'/'|'%') right=jassExpr
```
Comment (peq) [inline de.peeeq.wurstscript/src/main/antlr/de/peeeq/wurstscript/antlr/Jass.g4]: `%` is not valid in Jass

File: de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/jass/ExtendedJassLexer.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+import de.peeeq.wurstscript.antlr.JassParser;
+import de.peeeq.wurstscript.utils.LineOffsets;
+import org.antlr.v4.runtime.*;
+import org.antlr.v4.runtime.misc.Pair;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class ExtendedJassLexer implements TokenSource {
```
[the remark this answers is not in the document]
Comment (Frotty) [reply de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/jass/ExtendedJassLexer.java]: Okay good to know, I mostly copy & pasted this.

File: de.peeeq.wurstscript/src/main/antlr/de/peeeq/wurstscript/antlr/Jass.g4 (hunk 1 above)
Comment (Frotty) [reply de.peeeq.wurstscript/src/main/antlr/de/peeeq/wurstscript/antlr/Jass.g4]: The module operator `%` has been added with patch 1.29 according to @lep

**Pull request merged**